src/Fieldtypes/Entries.php

```diff
@@ -144,6 +144,20 @@ public function getIndexItems($request)
             $query->whereIn('blueprint', $blueprints);
         }
 
+        collect($this->getConfiguredCollections())
+            ->map(fn ($handle) => Collection::findByHandle($handle))
+            ->filter(fn ($collection) => User::current()->cant('view-other-authors-entries', [EntryContract::class, $collection]))
+            ->each(function ($collection) use ($query) {
+                $blueprintsWithoutAuthor = $collection->entryBlueprints()
+                    ->filter(fn ($blueprint) => ! $blueprint->hasField('author'))
+                    ->map->handle()->all();
+
+                    $query->where(fn ($query) => $query
```

Comment:
As this can be across multiple collections and blueprint names are potentially the same across multiple collections (but different blueprints) you should `->where('collection', $collction->handle())` here too.